311-status-protocol.md

```diff
@@ -0,0 +1,262 @@
+- Transport privacy layer (e.g. something like Loopix)
+- Conversational security layer (e.g. something like BTP)
+- Syncing layer (e.g. something like BSP)
+
+- Chat clients (public/private/group chat)
+
+### Team growth and collaboration with other teams
+This effort is about creating open specifications, so we need access to a wide
+range of specialized skill sets, as well as general consensus among community.
+Creating a protocol in isolation would likely lead to a lackluster result.
+
+This swarm will collaborate with Ethereum / Web3 / decentralized web / secure
+messaging teams in an attempt to re-use existing work and create protocols
+together.
+
+Additionally, this team will involve individuals with relevant expertise.
+Ideally individuals who have designed 1-2 open, battle-tested p2p protocols.
+
+### Reflection of our principles
+
+The protocol should be a [reflection](https://our.status.im/tag/our-principles/)
+of [our principles](https://our.status.im/our-principles/).
+
+- I. Liberty - enable sovereignty of individuals with things like key
+  management, as well as economic freedom, etc, by enabling transaction of funds
+  and socio-economic coordination between small groups of people.
+
+- II. Censorship resistance - enable censorship resistance through things like
+  pluggable transports, as well as being agnostic to the information being
+  transported.
+
+- III. Security - use state-of-the-art technologies, and research new security
+  methods and technologies to make strong security guarantees.
+
+- IV. Privacy - protect privacy in communication and transactions, and strive to
+  provide right to total anonymity.
+
+- V. Transparency - be open about what we are doing, allow community
+  contributions and be clear about trade-offs.
+
+- VI. Openness - specification is open and under a permissive license.
+
+- VII. Decentralization - p2p network, maximizing number of computers/humans who
+  can control and use the protocol we are building.
+
+- VIII. Inclusivity - easy to use our protocols, as well as working towards
+  interopability with other protocols, and educating users.
+
+- IX. Continuance - network should be incentivized to continue on its own.
+
+- X. Resourcefulness - work with other teams to avoid duplication of effort,
+  study prior work deeply to minimize wasted and duplicate effort.
+
+### Anonymous Communication Protocol requirements
+
+The purpose of an anonymous communication (AC) protocol is to ensure metadata
+isn’t leaked when messages are communicated between peers. It deals with how
+messages are transported, and not what is in them.
+
+This is the inital focus and work has begun together with Web3 Foundation and
+Validity Labs. Note that this list of requirements is tentative.
+
+1. Sender Anonymity
+2. Receiver Anonymity
+3. Sender-Receiver Unlinkability
+4. Reasonable Latency
+5. Reasonable Bandwidth
+6. Adaptable Anonymity
+7. Scalable
+8. No specialized Services
```

Comment:
I would still want to add that it should be designed with mobile in mind: internet connection is unstable an mostly missing, CPU resources are limited, we can't stay in background for a long time and we are behind very strict firewalls that makes it hard for NAT to pass through.

Maybe it is already incorporated in these requirements, but I just want to reiterate :-P

Reply:
Good point, https://github.com/w3f/messaging/issues/18